ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/RestApiTest.java

```diff
@@ -1039,6 +1042,106 @@ public void shouldCreateStreamWithVariableSubstitution() {
     assertThat(query.size(), is(1));
   }
 
+  @Test
+  public void shouldAssertTopicExists() {
+    // Given:
+    makeKsqlRequest("CREATE STREAM X AS SELECT * FROM " + PAGE_VIEW_STREAM + ";");
+
+    // When:
+    List<KsqlEntity> response = makeKsqlRequest("ASSERT TOPIC X WITH (PARTITIONS=1) TIMEOUT 2 SECONDS;");
+
+    // Then:
+    assertThat(response.size(), is(1));
+    assertThat(((AssertTopicEntity) response.get(0)).getTopicName(), is("X"));
+  }
+
+  @Test
+  public void shouldFailToAssertNonExistantTopic() {
+    assertThatEventually(() -> {
+      try {
+        makeKsqlRequest("ASSERT TOPIC X WITH (PARTITIONS=1) TIMEOUT 1 SECONDS;");
+        return "Should have thrown 'Topic does not exist' error.";
+      } catch (final Throwable t) {
+        return t.getMessage();
+      }
+    }, containsString("Topic X does not exist"));
+  }
+
+  @Test
+  public void shouldFailToAssertTopicWithWrongConfigs() {
+    // Given:
+    makeKsqlRequest("CREATE STREAM X AS SELECT * FROM " + PAGE_VIEW_STREAM + ";");
+
+    // When:
+    // Then:
+    assertThatEventually(() -> {
+      try {
+        makeKsqlRequest("ASSERT TOPIC X WITH (PARTITIONS='apples', REPLICAS=100, FAKE_CONFIG='Hello!') TIMEOUT 2 SECONDS;");
+        return "Should have thrown config mismatch error";
+      } catch (final Throwable t) {
+        return t.getMessage();
+      }
+    }, containsString(
+        "Mismatched configuration for topic X: For config partitions, expected apples got 1\n"
+            + "Mismatched configuration for topic X: For config replicas, expected 100 got 1\n"
```

Comment:
Is the expected text backwards?  The stream was created with (1,1) and (apples, 100) were offered, right?

Reply:
its `(the assert statement) expected 100, got 1 (from the topic instead)`

i can see how the wording is ambiguous. maybe something like would be clearer:

`For config partitions, 100 was asserted, but the actual value was 1`

Reply:
Ah, I read it backwards yesterday!